core/src/main/scala/kafka/server/ReplicaAlterLogDirsThread.scala

```diff
@@ -208,7 +208,8 @@ class ReplicaAlterLogDirsThread(name: String,
         requestMap.put(topicPartition, new FetchRequest.PartitionData(partitionFetchState.fetchOffset, logStartOffset,
           fetchSize, Optional.of(partitionFetchState.currentLeaderEpoch)))
       } catch {
-        case _: KafkaStorageException =>
+        case e: KafkaStorageException =>
+          info(s"Failed to build fetch for $topicPartition with $e")
```

Comment:
I'm not sure we get much benefit from this message. It seems the only way `KafkaStorageException` will be raised from `futureLocalReplicaOrException` is if we had already caught a disk error and marked the partition offline. Perhaps debug level would be reasonable?

By the way, the exception can be passed as an argument rather than embedded in the message.